app/imageswap/test/test_image_patterns.py

```diff
@@ -0,0 +1,214 @@
+#!/usr/bin/env python
+
+# Copyright 2020 The WebRoot, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys
+import unittest
+from unittest.mock import patch
+
+sys.path.append("./app/imageswap")
+import imageswap
+
+###########################################################################
+# Test image definition/syntax scenarios ##################################
+###########################################################################
+
+
+@patch("imageswap.imageswap_mode", "MAPS")
+@patch("imageswap.imageswap_maps_file", "./testing/map_files/map_file.conf")
+class ImageFormats(unittest.TestCase):
+    def setUp(self):
+
+        self.app = imageswap.app.test_client()
+        self.app.testing = True
+        imageswap.app.logger.setLevel("DEBUG")
+
+    def tearDown(self):
+
+        pass
+
+    def test_image_format_image_only(self):
+
+        """Method to test MAP based swap (image only: \"alpine\")"""
+
+        container_spec = {}
+        container_spec["name"] = "test-container"
+        container_spec["image"] = "alpine"
+
+        expected_image = "my.example.com/mirror-docker.io/alpine"
+        result = imageswap.swap_image(container_spec)
+
+        self.assertTrue(result)
+        self.assertEqual(container_spec["image"], expected_image)
+
+    def test_image_format_image_tag(self):
+
+        """Method to test MAP based swap (image+tag: \"nginx:latest\")"""
+
+        container_spec = {}
+        container_spec["name"] = "test-container"
+        container_spec["image"] = "nginx:latest"
+
+        expected_image = "my.example.com/mirror-docker.io/nginx:latest"
+        result = imageswap.swap_image(container_spec)
+
+        self.assertTrue(result)
+        self.assertEqual(container_spec["image"], expected_image)
+
+    def test_image_format_project_image(self):
+
+        """Method to test MAP based swap (project+image: \"nginx:latest\")"""
```

Comment:
**Suggestion:** Looks like maybe a typo in this output. Says "nginx:latest" but then test used "ubuntu/ubuntu".

Reply:
Fixed

Reply:
Thanks!